os/arch/arm/src/armv7-r/arm_assert.c

```diff
@@ -918,6 +931,67 @@ static void up_dumpstate(void)
 static void _up_assert(int errorcode) noreturn_function;
 static void _up_assert(int errorcode)
 {
+#ifdef CONFIG_FAULT_MGR
+	int ret;
+	mqd_t fault_queue;
+	pid_t pid;
+	uint32_t assert_pc;
+	uint32_t ksram_segment_end = (uint32_t)__ksram_segment_start__ + (uint32_t)__ksram_segment_size__;
+	uint32_t kflash_segment_end = (uint32_t)__kflash_segment_start__ + (uint32_t)__kflash_segment_size__;
+
+	/* Extract the PC value of instruction which caused the abort/assert */
+
+	if (current_regs) {
+		assert_pc = current_regs[REG_LR];
+	} else {
+		assert_pc = (uint32_t)g_assertpc[0];
+	}
+
+	/* Check if the PC lies within kernel context */
+
+	if ((assert_pc >= (uint32_t)__ksram_segment_start__ && assert_pc <= ksram_segment_end) || (assert_pc > (uint32_t)__kflash_segment_start__ && assert_pc < kflash_segment_end)) {
+
+		/* Reboot the system if fault is in kernel context */
+
+#ifdef CONFIG_BOARDCTL_RESET
+		boardctl(BOARDIOC_RESET, EXIT_SUCCESS);
+#else
+		for (;;) {
+#ifdef CONFIG_ARCH_LEDS
+			board_autoled_on(LED_PANIC);
+			up_mdelay(250);
+			board_autoled_off(LED_PANIC);
+			up_mdelay(250);
+#endif
+		}
+#endif
+	} else {
+		fault_queue = mq_open(FAULT_MGR_MQ, O_WRONLY | O_CREAT, 0666, 0);
```

Comment:
The queue should not be created here...since it is already opened during bootup time.

Reply:
okay, will modify it.